dawn/prototype/AtlasFromNetcdf.cpp

```diff
@@ -0,0 +1,298 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "AtlasFromNetcdf.h"
+
+#include <assert.h>
+#include <iostream>
+#include <math.h>
+
+#include <netcdf>
+
+#include <atlas/array/Array.h>
+#include <atlas/array/ArrayView.h>
+#include <atlas/array/IndexView.h>
+#include <atlas/library/Library.h>
+#include <atlas/library/config.h>
+#include <atlas/mesh/ElementType.h>
+#include <atlas/mesh/Elements.h>
+#include <atlas/mesh/HybridElements.h>
+#include <atlas/mesh/Nodes.h>
+#include <atlas/mesh/actions/BuildCellCentres.h>
+#include <atlas/mesh/actions/BuildEdges.h>
+#include <atlas/mesh/actions/BuildPeriodicBoundaries.h>
+#include <atlas/meshgenerator/detail/MeshGeneratorImpl.h>
+#include <atlas/output/Gmsh.h>
+#include <atlas/util/CoordinateEnums.h>
+
+static const int myPart = 0;
+
+template <typename loadT>
+static std::vector<loadT> LoadField(const netCDF::NcFile& dataFile, const std::string& name) {
+  netCDF::NcVar data = dataFile.getVar(name.c_str());
+  if(data.isNull()) {
+    return {};
+  }
+  assert(data.getDimCount() == 1);
+  int size = data.getDim(0).getSize();
```

Comment:
It returns `size_t`. Changed this here, but I'm converting implicitly to `int` later because those values are used as loop boundaries and I have been burned by stuff like `for (size_t i = 0; i < v.size() - 1; i++) { }`  too often.

Reply:
I wouldn't do that. That would be a narrowing conversion which has other problems, even harder to spot.